.github/workflows/on-master-commit.yaml

```diff
@@ -0,0 +1,48 @@
+name: Master branch CI
+
+on:
+  push:
+    branches:
+      - 'master'
+
+jobs:
+  run-unit-tests:
+    name: Run Unit Tests
+    uses: ./.github/workflows/test.yaml
+
+  lint:
+    name: Lint
+    uses: ./.github/workflows/lint-go.yaml
```

Comment:
Optional optimization: These could be made parallel.

Reply:
I think its better if they don't as the tests could fail but we end up publishing the docker image anyway

Reply:
I see, I mean just the lint and test as a parallel step. Albeit, in this flow it won't make much difference given that the lint is pretty fast.